src/core/dbus-timer.c

```diff
@@ -236,7 +236,7 @@ static int bus_timer_set_transient_property(
                 if (!UNIT_WRITE_FLAGS_NOOP(flags)) {
                         r = calendar_spec_from_string(str, &c);
                         if (r < 0)
-                                return r;
+                                return sd_bus_error_setf(error, SD_BUS_ERROR_INVALID_ARGS, "Invalid calendar spec");
```

Comment:
returning prettier messages definitely makes a lot of sense, but strictly speaking calendar_spec_from_string() might return other errors than just EINVAL (for example, ENOMEM). Hence I think this should be changed to:

```c
if (r == -EINVAL)
        return sd_bus_error_setf(error, …
if (r < 0)
         return r;
```

i.e. catch the relevant error explicitly, and return all others as-is